### PR TITLE
Upgrade node version to fix local testing

### PR DESCRIPTION
**Pull request by snowystinger**
update our executors to a common set of definitions
update to node 15, this is needed to use some unit tests involving numberformat

Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/i18n/test/numberFormatSignDisplayPolyfill.test.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function verify(locale, options, signDisplay, v) {
   let a = new Intl.NumberFormat(locale, options);
   let b = new Intl.NumberFormat(locale, {...options, signDisplay});
 
-  expect(b.format(v)).toBe(numberFormatSignDisplayPolyfill(a, signDisplay, v));
+  expect(numberFormatSignDisplayPolyfill(a, signDisplay, v)).toBe(b.format(v));
```
Comment (ktabors) [inline packages/@react-aria/i18n/test/numberFormatSignDisplayPolyfill.test.js:19]: Why does changing the order fix this?

**Review by ktabors — APPROVED** [review]

File: packages/@react-aria/i18n/test/numberFormatSignDisplayPolyfill.test.js (hunk 1 above)
Comment (snowystinger) [reply packages/@react-aria/i18n/test/numberFormatSignDisplayPolyfill.test.js:19]: it doesn't fix it, it makes the message output on failure easier to read

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+executors:
+  default-executor:
+    docker:
+      - image: cimg/node:15.3
+    working_directory: ~/react-spectrum
+
+  large-executor:
     docker:
-      - image: circleci/node:12
+      - image: cimg/node:15.3
```
Comment (devongovett) [inline .circleci/config.yml:13]: Why are we updating node for react 17 tests? I'm still using 12 locally? Will I also need to upgrade? And what is cimg?

File: packages/@react-aria/i18n/src/utils.ts [hunk 3]
```diff
@@ -50,14 +50,12 @@ export function numberFormatSignDisplayPolyfill(numberFormat: Intl.NumberFormat,
     }
 
     if (needsPositiveSign) {
-      let negative = numberFormat.format(-num);
```
Comment (devongovett) [inline packages/@react-aria/i18n/src/utils.ts:53]: Was this broken in react 17 stable? Why the changes?

File: .circleci/config.yml (hunk 2 above)
Comment (snowystinger) [reply .circleci/config.yml:13]: Looks like i conflated two issues
one is the node upgrade to fix a failing test, the other is to upgrade react 17, i'll split it out

as for upgrading node, yes, but you should be using nvm, so it's an easy thing to do

Comment (snowystinger) [reply .circleci/config.yml:13]: cimg is their new way of naming circleci images

Comment (snowystinger) [reply .circleci/config.yml:13]: waiting for this to build and make sure everything is good before removing the react 17 upgrade stuff from this branch https://github.com/adobe/react-spectrum/pull/1354

File: packages/@react-aria/i18n/src/utils.ts [hunk 4]
```diff
@@ -50,14 +50,12 @@ export function numberFormatSignDisplayPolyfill(numberFormat: Intl.NumberFormat,
...
-      // ignore RTL/LTR marker character
-      let minus = negative.replace(noSign, '').replace(/\u200e|\u061C/, '');
-      if ([...minus].length !== 1) {
+      let minus = numberFormat.formatToParts(-num).find(part => part.type === 'minusSign').value;
+      if (minus.length === 0) {
         console.warn('@react-aria/i18n polyfill for NumberFormat signDisplay: Unsupported case');
       }
-      let positive = negative.replace(noSign, '!!!').replace(minus, '+').replace('!!!', noSign);
+      let negative = numberFormat.format(-num);
+      let positive = negative.replace(minus, '+');
```
Comment (LFDanLu) [inline packages/@react-aria/i18n/src/utils.ts:58]: Just wanna make sure that I understand this code block above: essentially this adds a '+' sign in from of the number except if the current numberFormat doesn't allow displaying signs?

Comment (LFDanLu) [reply packages/@react-aria/i18n/src/utils.ts:58]: Looks good to me otherwise

Comment (snowystinger) [reply packages/@react-aria/i18n/src/utils.ts:58]: exceptZero is an option not supported everywhere, so what this does is, if it's a positive number, gets the formatted negative number equivalent and the actual minusSign character. Then it replaces the minus sign character with the positive sign.

**Review by LFDanLu — APPROVED** [review]

**Comment by devongovett** [comment]
Hmm, I don't think we should require Node 15 yet. 14 is the current LTS and 12 is supported until 2022. Can we figure out why it passes in CI but not locally?

**Comment by snowystinger** [comment]
Ah yep, I looked at the graph, didn't read the table below it https://nodejs.org/en/about/releases/ yeah, we can look more into why it's failing locally
